frontend/src/libs/components/article-form/article-form.tsx

```diff
@@ -0,0 +1,103 @@
+import { ButtonType } from '~/libs/enums/button-type.enum.js';
+import { useAppDispatch, useAppForm, useCallback } from '~/libs/hooks/hooks.js';
+import {
+  type ArticleCreateRequestDto,
+  articleCreateValidationSchema,
+} from '~/packages/articles/articles.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
+
+import { Button, Input, Textarea } from '../components.js';
+import styles from './styles.module.scss';
+
+const DEFAULT_ARTICLE_FORM_PAYLOAD: ArticleCreateRequestDto = {
+  title: '',
+  text: '',
+  genreId: 1,
+};
+// replace to src/pages/article/components/article-form/libs/constant.ts
+//when Article component will be implemented
+
+const ArticleForm: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { control, errors, handleSubmit, reset } =
+    useAppForm<ArticleCreateRequestDto>({
+      defaultValues: DEFAULT_ARTICLE_FORM_PAYLOAD,
+      validationSchema: articleCreateValidationSchema,
+    });
+
+  const handleArticleSubmit = useCallback(
+    (buttonName: string) =>
+      (payload: ArticleCreateRequestDto): void => {
+        const updatedPayload = {
+          ...payload,
+          ...(buttonName === 'publish' && {
+            publishedAt: new Date().toISOString(),
+          }),
+        };
+
+        void dispatch(articlesActions.createArticle(updatedPayload));
```

Comment:
Split it into 2 different actions - `publishArcticle` and `saveArticleDraft`,, and let's not prepopulate publishedAt from FE

Reply:
If I don't pass publishedAt from the FE, then how do I distinguish whether it's a save or a publication? My logic is as follows: saving and publication follow the same route, if the field is in the request (I expect the date in ISO 8601 format from the FE), then if nothing interferes (validation of other fields), then it writes the data to the publishedAt field.

Reply:
@andanf-e 
Maybe we misunderstood?

Once again, the way I see it, and by the way, if I remember correctly, it was originally described in the task

1. when creating a publication, the publishedAt field is optional. if it is, then it is a publication. If not, it is a save.
2. When updating, I also provided the ability to pass the publishedAt field as optional, and I think it can be passed to null if, for example, the user wants to remove an article from published.

Or there is another option, to make this logic, all through the same routes, but to pass some options through query parameters.

@OlenaMohylivska29, @andanf-e, @nickolay-grechkin, @WilhelmWesser Let's discuss it, share your vision and ideas?

Reply:
One more option is to pass the extra header  XActionType: 'publish', but I am inclined to use query params too

Reply:
Yep, sorry
I've overthought it a little bit -- let's keep it as it is with one route and publishedAt payload